components/mpas-framework/src/framework/mpas_pool_routines.F

```diff
@@ -1014,83 +1032,135 @@ recursive subroutine mpas_pool_copy_pool(srcPool, destPool)!{{{
                   else if (associated(dptr % l0)) then
                      call mpas_duplicate_field(dptr % l0, mem % l0, copy_array_only=.true.)
                   else if (associated(dptr % r0a)) then
-                     do j=1,mem % contentsTimeLevs
-                        mem % r0 => mem % r0a(j)
-                        call mpas_duplicate_field(dptr % r0a(j), mem % r0, copy_array_only=.true.)
-                        nullify(mem % r0)
-                     end do
+                     if (timeLevels > 1) then
+                        do j=1,timeLevels
+                           mem % r0 => mem % r0a(j)
+                           call mpas_duplicate_field(dptr % r0a(j), mem % r0, copy_array_only=.true.)
+                           nullify(mem % r0)
+                        end do
+                     else
+                         call mpas_duplicate_field(dptr % r0a(1), mem % r0, copy_array_only=.true.)
+                     end if
```

Comment:
I'm a little confused why the `timeLevels==1` case is handled separately (here and throughout below) when it wasn't before.

Reply:
@matthewhoffman, prior to these changes, the `mpas_pool_copy_pool` had assumed that the src/dest pools had the same number of time levels. However, for RK4 the state pool has 2 time levels, but we only need to copy the first into the provis pool. The special treatment is necessary since for a single time level pool, data is stored in the r0/r1/r2/r3... fields as opposed to the r0a/r1a/r2a/r3a... fields where data is stored for pools with >1 timelevels. So, there needs to be a way to copy r0a(1)/r1a(1)/r2a(1)/r3a(1) into r0/r1/r2/r3 when the override time levels option is present. `mpas_pool_clone_pool` (which was what was originally used for this purpose state->provis copy) has this some functionality and I implemented this as close to how that subroutine works as possible.

Reply:
@sbrus89 , thanks for the explanation!